app/frontend/components/domains/requirements-library/requirement-field-edit/conditional-setup-modal.tsx

```diff
@@ -15,25 +18,119 @@ import {
+import { RequirementSelect } from "../../../shared/select/selectors/requirement-select"
+import { IRequirementBlockForm } from "../requirements-block-modal"
 
 interface IProps {
   triggerButtonProps?: Partial<ButtonProps>
   renderTriggerButton?: (props: ButtonProps) => JSX.Element
-  conditional: IRequirementAttributes["inputOptions"]["conditional"]
+  index: number
 }
 
-export function ConditionalSetupModal({ triggerButtonProps, renderTriggerButton, conditional }: IProps) {
+export const ConditionalSetupModal = observer(({ triggerButtonProps, renderTriggerButton, index }: IProps) => {
   const { isOpen, onOpen, onClose } = useDisclosure()
   const { t } = useTranslation()
+
+  const formMethods = useFormContext<IRequirementBlockForm>()
+
+  const { control, watch, setValue } = formMethods
+
+  const watchedLabel = watch(`requirementsAttributes.${index}.label`)
+
+  const watchedWhen = watch(`requirementsAttributes.${index}.inputOptions.conditional.when`)
+  const watchedOperand = watch(`requirementsAttributes.${index}.inputOptions.conditional.operand`)
+  const watchedThen = watch(`requirementsAttributes.${index}.inputOptions.conditional.then`)
+  const allFieldsProvided = watchedWhen && watchedOperand && watchedThen
+
+  const watchedRequirements = watch(`requirementsAttributes`)
+  const watchedRequirementCode = watch(`requirementsAttributes.${index}.requirementCode`)
+  const selectedRequirementAttr = watchedRequirements?.find((reqAttr) => reqAttr.requirementCode === watchedWhen)
+
+  const [showAdvanced, setShowAdvanced] = useState<boolean>(false)
+  const toggleShowAdvanced = () => setShowAdvanced((cur) => !cur)
+
+  const [operandOptions, setOperandOptions] = useState<IOption[]>(null)
+  const [requirementOptions, setRequirementOptions] = useState<IOption[]>(null)
+
+  const onReset = () => {
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.when`, null)
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.operand`, null)
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.then`, null)
+  }
+
+  const getOperandOptions = () => {
+    if (selectedRequirementAttr?.inputOptions?.valueOptions) {
+      return selectedRequirementAttr?.inputOptions.valueOptions
+    }
+    if (selectedRequirementAttr?.inputType === "checkbox") {
+      return [
+        { label: t("ui.checked"), value: "true" },
+        { label: t("ui.unchecked"), value: "false" },
+      ]
+    }
+    return []
+  }
+
+  const getRequirementOptions = () => {
+    return watchedRequirements
+      .map((requirement) => ({ label: requirement.label, value: requirement.requirementCode }))
+      .filter((opt) => opt.value !== watchedRequirementCode)
+  }
+
+  useEffect(() => {
+    setRequirementOptions(getRequirementOptions())
+    if (watchedWhen) {
+      setOperandOptions(getOperandOptions())
+    }
+  }, [watchedWhen])
+
+  const effectOptions = ["show", "hide"].map((value) => ({
+    value,
```

Comment:
![Screenshot 2024-04-26 at 10 26 03 AM](https://github.com/bcgov/HOUS-permit-portal/assets/32022335/78b83252-4faa-4a55-8c32-c093c453da3f)
Using the following, should allow you to remove `ts-ignore`